packages/polling-controller/src/StaticIntervalPollingController.ts

```diff
@@ -30,8 +30,22 @@ function StaticIntervalPollingControllerMixin<
   {
     readonly #intervalIds: Record<PollingTokenSetId, NodeJS.Timeout> = {};
 
+    /**
+     * A map of polling token set IDs to their durations.
+     * This is used to determine when to stop polling for a specific token set.
+     */
+    #durationsById: Record<PollingTokenSetId, number> = {};
+
     #intervalLength: number | undefined = 1000;
 
+    setDurationForId(key: string, duration: number) {
```

Comment:
Hmm, I'm wondering if we can come up with a different API. This feels a bit awkward to me IMO. It seems that you would first need to generate the key that `startPolling` already generates for you so you can call `setDurationForId` first and then `startPolling`. 

Instead of adding a separate setter and getter for the duration, what if you could pass in a `duration` to `startPolling`? So you're saying "I want to start a poll that will last this long". And then it just takes care of it for you.

I realize this may require some extra API changes, I can look into this a bit further, but I am curious to know your thoughts.